test/e2e/vsn.js

```diff
@@ -0,0 +1,242 @@
+const tapePromise = require('tape-promise').default
+const tape = require('tape')
+const isRoughlyEqual = require('is-roughly-equal')
+
+const {createWhen} = require('./lib/util')
+const createClient = require('../..')
+const vsnProfile = require('../../p/vsn')
+const products = require('../../p/vsn/products')
+const {
+	line: createValidateLine,
+	journeyLeg: createValidateJourneyLeg,
+	movement: _validateMovement
+} = require('./lib/validators')
+const createValidate = require('./lib/validate-fptf-with')
+const testJourneysStationToStation = require('./lib/journeys-station-to-station')
+const testJourneysStationToAddress = require('./lib/journeys-station-to-address')
+const testJourneysStationToPoi = require('./lib/journeys-station-to-poi')
+const testEarlierLaterJourneys = require('./lib/earlier-later-journeys')
+const journeysFailsWithNoProduct = require('./lib/journeys-fails-with-no-product')
+const testDepartures = require('./lib/departures')
+const testArrivals = require('./lib/arrivals')
+
+const when = createWhen('Europe/Berlin', 'de-DE')
+
+const cfg = {
+	when,
+	products,
+	minLatitude: 47.24,
+	maxLatitude: 52.9,
+	minLongitude: -0.63,
+	maxLongitude: 14.07
+}
+
+const _validateLine = createValidateLine(cfg)
+const validateLine = (validate, l, name) => {
+	if (!l.direction) l = Object.assign({}, l, {direction: 'foo'})
+	_validateLine(validate, l, name)
+}
+
+const _validateJourneyLeg = createValidateJourneyLeg(cfg)
+const validateJourneyLeg = (validate, l, name) => {
+	if (!l.direction) l = Object.assign({}, l, {direction: 'foo'})
+	_validateJourneyLeg(validate, l, name)
+}
+
+const validateMovement = (val, m, name = 'movement') => {
+	// todo: fix this upstream
+	const withFakeLocation = Object.assign({}, m)
+	withFakeLocation.location = Object.assign({}, m.location, {
+		latitude: 50,
+		longitude: 12
+	})
+	_validateMovement(val, withFakeLocation, name)
+
+	assert.ok(m.location.latitude <= 55, name + '.location.latitude is too small')
+	assert.ok(m.location.latitude >= 45, name + '.location.latitude is too large')
+	assert.ok(m.location.longitude >= 1, name + '.location.longitude is too small')
+	assert.ok(m.location.longitude <= 11, name + '.location.longitude is too small')
+}
+
+const validate = createValidate(cfg, {
+	line: validateLine,
+	journeyLeg: validateJourneyLeg,
+	movement: validateMovement
+})
+
+const test = tapePromise(tape)
+const client = createClient(vsnProfile, 'public-transport/hafas-client:test')
+
+const kornmarkt = '9033977'
+const jugendherberge = '9033961'
+const ewaldstrasse = '9033896'
+
+test('journeys – Kornmarkt to Ewaldstraße', async (t) => {
+	const res = await client.journeys(kornmarkt, ewaldstrasse, {
+		results: 4,
+		departure: when,
+		stopovers: true
+	})
+
+	await testJourneysStationToStation({
+		test: t,
+		res,
+		validate,
+		fromId: kornmarkt,
+		toId: ewaldstrasse
+	})
+	t.end()
+})
+
+// todo: journeys, only one product
+
+test('journeys – fails with no product', (t) => {
```

Comment:
Asserting this as part of the E2E tests doesn't make sense, it's a leftover of the crufty test code that I mentioned. You can remove it.